### PR TITLE
improve error handling

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

## Context 

Follow up of https://github.com/giantswarm/aws-operator/pull/2785. We have the same error handling problem here now as well. Nobody knows why that happens now.

### CODE REVIEW

File: service/internal/tenantclient/client.go [hunk 1]
```diff
@@ -83,11 +82,10 @@ func (c *TenantClient) K8sClient(ctx context.Context, obj interface{}) (k8sclien
 		}
 
 		k8sClient, err = k8sclient.NewClients(c)
-		if tenant.IsAPINotAvailable(err) {
-			return nil, microerror.Mask(notAvailableError)
-		} else if err != nil {
-			return nil, microerror.Mask(err)
+		if err != nil {
+			return nil, microerror.Maskf(notAvailableError, err.Error())
```
Comment (xh3b4sd) [inline service/internal/tenantclient/client.go:86]: This here is the actual "fix". On any error we act accordingly. The error matching and graceful handler cancelation is already in place everywhere.

**Review by njuettner — APPROVED** [review]
lgtm

**Review by anvddriesch — APPROVED** [review]

File: pkg/project/project.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ var (
 	gitSHA      = "n/a"
 	name        = "cluster-operator"
 	source      = "https://github.com/giantswarm/cluster-operator"
-	version     = "3.3.1-dev"
+	version     = "3.3.1-xh3b4sd"
```
Comment (xh3b4sd) [inline pkg/project/project.go:8]:
```suggestion
	version     = "3.3.1-dev"
```

**Comment by xh3b4sd** [comment]
With this I could create Tenant Clusters again.

**Pull request merged**